test/extract-js.js

```diff
@@ -0,0 +1,71 @@
+var assert = require('assert');
+var Checker = require('../lib/checker');
+
+describe('modules/extract', function() {
+    describe('common', function() {
+        it('should not throw an exceptions on error explain (console reporter)', function() {
+            var checker = new Checker();
+            checker.registerDefaultRules();
+            checker.configure({
+                disallowKeywords: ['with'],
+                extract: 'always'
+            });
+
+            return checker.checkPath('./test/data/extract').then(function(errorsCollection) {
+                var errorCount = 0;
+
+                assert.doesNotThrow(function() {
+                    errorsCollection.forEach(function(errors) {
+                        if (!errors.isEmpty()) {
+                            errors.getErrorList().forEach(function(error) {
+                                errors.explainError(error, true);
```

Comment:
Ah... I just remembered! Reporter crashes as we use custom file class (`HtmlFile`) in errors. So error details should be compatible with console reporter (other reporters ok, but console reporter not, because do additional formatting). Something like that.